services/user-services.js

```diff
@@ -0,0 +1,46 @@
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+
+const userDao = require('../models/user.dao');
+const {checkEmail}  = require('../validators');
+
+const signUp = async(email, password, firstname, lastname, skintype) => {
+ 
+  await checkEmail(email)
+  const user = await userDao.connectuser(email);
+    if(user){
+    const err = new Error('duplicated email');
+    err.statusCode = 400;
+    throw err;
+  }
+  const hashedPassword = await bcrypt.hash(password, 10);
+  await userDao.createuser(email, hashedPassword, firstname, lastname, skintype); 
+};
+
+const signIn = async ( email, password) => {
+  
+  const user = await userDao.connectuser(email);
+
+  if (!user) {
+    const err = new Error('user does not exist');
+    err.statusCode = 404;
+    throw err;
+  }
+  
+  const result = await bcrypt.compare(password, user.password);
+
+  if (!result) {
+    const err = new Error('invalid password');
+    err.statusCode = 401;
+    throw err;
+  }
+
+  return jwt.sign({ sub: user.id, email:user.email }, process.env.JWT_SECRET);
```

Comment:
jwt의 payload에는 유저의 민감한 정보를 포함하지 않습니다.
민감한 정보라고 한다면 다음과 같습니다:
- 유저의 이메일
- 유저의 비밀번호
- 유저의 이름
등 유저에 대한 개인 정보를 포함합니다.

우리는 데이터베이스의 테이블에서 유저를 특정할 수 있는 고유 값이면서 아무 의미가 없는 숫자로 이루어진 데이터를 가지고 있습니다.
바로 id 값입니다.

지금처럼 jwt 모듈에서 기본적으로 제공하고 있는 claim에 유저의 `id`를 포함하는 것도 좋지만, custom cliam을 새로 설정할 수 있습니다. 아래 코드를 참고해주세요.

아래 문서를 통해 확인해보면 더 좋을 것 같습니다.
https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-token-claims

```suggestion
  return jwt.sign({ userId: user.id }, process.env.JWT_SECRET);
```